File: GO/Cap04/c04ex06.go
// c04ex06.go

package main

import "fmt"

func main() {
  var A, B, Opcao float64

  fmt.Print("Entre o 1o. valor numerico: ")
  fmt.Scan(&A)

  fmt.Print("Entre o 2o. valor numerico: ")
  fmt.Scan(&B)

  fmt.Println()
  fmt.Println("Escolha uma opcao de menu:")
  fmt.Println()
  fmt.Println("[1] - Adicao")
  fmt.Println("[2] - Subtracao")
  fmt.Println("[3] - Multiplicacao")
  fmt.Println("[4] - Divisao")
  fmt.Println()

  fmt.Print("==> ")
  fmt.Scan(&Opcao)
  fmt.Println()

  switch (Opcao) {
    case 1:
      fmt.Println("Calculo de adicao")
      fmt.Println("Resultado: ", A + B)
    case 2:
      fmt.Println("Calculo de subtracao")
      fmt.Println("Resultado: ", A - B)
    case 3:
      fmt.Println("Calculo de multiplicacao")
      fmt.Println("Resultado: ", A * B)
    case 4:
      fmt.Println("Calculo de divisao")
      if (B == 0) {
        fmt.Println("Resultado: ERRO")
      } else {
        fmt.Println("Resultado: ", A / B)
      }
    default:
      fmt.Println("Entrada invalida")
      fmt.Println("Tente novamente em outro momento")
  }

  fmt.Println()
  fmt.Print("Tecle <Enter> para encerrar... ")
  fmt.Scanln()
}

File: SWIFT/Cap04/c04ex06.swift
// c04ex06.swift

import Foundation

print("Entre o 1o. valor numerico: ", terminator: "")
guard let inputA = readLine(), let a = Float(inputA) else {
    fatalError("Entrada invalida.")
}

print("Entre o 2o. valor numerico: ", terminator: "")
guard let inputB = readLine(), let b = Float(inputB) else {
    fatalError("Entrada invalida.")
}

print()
print("Escolha uma opcao de menu:")
print()
print("[1] - Adicao")
print("[2] - Subtracao")
print("[3] - Multiplicacao")
print("[4] - Divisao")
print()

print("==> ", terminator: "")
guard let inputOpcao = readLine(), let opcao = Int(inputOpcao) else {
    fatalError("Entrada invalida.")
}

print()

switch (opcao) {
    case 1:
        print("Calculo de adicao")
        print("Resultado: \(String(format: "%.2f", (a + b)))") 
    case 2:
        print("Calculo de subtracao")
        print("Resultado: \(String(format: "%.2f", (a - b)))")
    case 3:
        print("Calculo de multiplicacao")
        print("Resultado: \(String(format: "%.2f", (a * b)))")
    case 4:
        print("Calculo de divisao")

        if (b == 0) {
            print("Resultado: ERRO")
        } else {
            print("Resultado: \(String(format: "%.2f", (a / b)))")
        }
    default:
        print("Entrada invalida")
        print("Tente novamente em outro momento")
}

print()
print("Tecle <Enter> para encerrar... ", terminator: "")
_ = readLine()
